src/vendorDataManager.ts

```diff
@@ -0,0 +1,32 @@
+import { removeCookie, storage } from "@guardian/libs";
+import { getConsentFor } from "./getConsentFor";
+import { onConsentChange } from "./onConsentChange";
+import { VendorWithCookieData, VendorWithLocalStorageData, vendorCookieData, vendorLocalStorageData } from "./vendorStorageIds";
+
+/**
+ * This function is called when the CMP is initialised. It listens for consent changes and removes cookies and localStorage data for vendors that the user has not consented to.
+ */
+export const initVendorDataManager = (): void => {
+	onConsentChange((consent) => {
+		(<VendorWithCookieData[]>Object.keys(vendorCookieData)).forEach((vendor) => {
+			const consentForVendor = getConsentFor(vendor, consent);
+			if (!consentForVendor) {
+				vendorCookieData[vendor].forEach((name) => {
+					removeCookie({name});
+				});
+
+			}
+		});
+
+		(<VendorWithLocalStorageData[]>Object.keys(vendorLocalStorageData)).forEach((vendor) => {
+			const consentForVendor = getConsentFor(vendor, consent);
```

Comment:
do we need to run through these loops twice? as in, i think these loops are the same and the `if (!consentForVendor)` check the same?

Reply:
They're two different loops one for localstorage, one for cookies, could use a single loop but I think it would require storing the localstorage/cookie keys in a less idiomatic way.